idx/bigtable/bigtable.go

```diff
@@ -533,3 +539,38 @@ func (b *BigtableIdx) prune() {
 		}
 	}
 }
+
+// AddDefs adds defs to the index.
+func (b *BigtableIdx) AddDefs(defs []schema.MetricDefinition) {
+	pre := time.Now()
+
+	b.MemoryIndex.AddDefs(defs)
+
+	if b.cfg.UpdateBigtableIdx {
+		// Blocking write to make sure all get enqueued
+		for _, def := range defs {
+			b.writeQueue <- writeReq{recvTime: time.Now(), def: &def}
+		}
+	}
+
+	statControlRestoreDuration.Value(time.Since(pre))
+}
+
+// DeleteDefs deletes the matching key.
+func (b *BigtableIdx) DeleteDefs(defs []schema.MetricDefinition, archive bool) {
+	pre := time.Now()
+
+	b.MemoryIndex.DeleteDefs(defs, archive)
+
+	if b.cfg.UpdateBigtableIdx {
+		// TODO - this could create a lot of go routines if many deletes come in at the same time
+		// Maybe better to enhance the write queue to process these?
```

Comment:
we have [briefly discused consistency and sync vs async](https://github.com/grafana/metrictank/issues/1976#issuecomment-865870923)
I agree it's sensible to create goroutines (as to not stall ingestion), but one goroutine per batch doesn't seem excessive.
My main concern is overloading bigtable because this "escapes" our concurrency restrictions. a queue seems appropriate.

There's also the more theoretical concern that these deletes should be executed within the kafka retention interval, although in practice that should always be true. (IIRC we recommend always replaying the kafka backlog and having kafka retention of many hours)